### PR TITLE
Add all code from private repository

### PR DESCRIPTION
**Pull request by arxeiss**
This is copy from SDK folder. Performed changes:
- Changed Go module name to reflect repository name
- Added more linter rules

This is just to add all code to repo. Further refactoring and publishing will be done in future sprints.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,10 @@
-# opa-indykite-plugin
+# IndyKite Identity Agent
+
+IndyKite Identity Agent extends the Open Policy Agent with features from
+the Identity Platform.
+
+## Invoke version
+
+```shell script
+ docker container run -it  indykite/indy-agent:latest version
```
Comment (bakkelim) [inline README.md:9]: should this be `indykite/opa:latest`?
Atleast that's what's it called in `.github/workflows/build-and-publish.yaml` and in dockerhub (public repo)

File: cmd/root.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+// RootCommand represents the OPA CLI Command.
+var RootCommand = ocmd.RootCommand
+
+func init() {
+	RootCommand.Short = "Indy Open Policy Agent (OPA)"
+	RootCommand.Long = `Indy Agent is a client side
+
+	TODO fix the description
+	`
+}
```
Comment (bakkelim) [inline cmd/root.go:31]: are we using this file?

File: config.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+OFFstatus:
+    service: acmecorp
+
+default_decision: /http/example/authz/allow
+
+keys:
+    global_key:
+        algorithm: RS256
+        key: <PEM_encoded_public_key>
```
Comment (bakkelim) [inline config.yaml:58]: is this entire file okay, in regards to this repo being public?

File: main.go [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+	"github.com/indykite/opa-indykite-plugin/cmd"
+
+	_ "github.com/indykite/opa-indykite-plugin/functions"
+	_ "github.com/indykite/opa-indykite-plugin/plugins"
+)
+
+func main() {
+	if err := cmd.RootCommand.Execute(); err != nil {
+		fmt.Println(err)
```
Comment (bakkelim) [inline main.go:30]: is this command needed to initialize plugins?

File: functions/client.go [hunk 5]
```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2020 IndyKite
```
Comment (marvinkite) [inline functions/client.go:1]: Let's update all date to 2022

File: functions/client.go [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+	clientMtx  sync.Mutex
+)
+
+// OverrideClient set and override Client Connection for OPA and return previously set connection.
+func OverrideClient(conn *identity.Client) *identity.Client {
+	oldConn := clientConn
+	clientConn = conn
+	return oldConn
+}
+
```
Comment (marvinkite) [inline functions/client.go:40]: Let's document all public functions. This is a function others can use to create new functions and extend the plugin.

File: functions/identity.go [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+
+			// ast.ObjectTerm(k)
+			//
+			//
+			// v, err := ast.ValueFromReader(resp.Body)
+			// if err != nil {
+			//	return nil, err
+			// }
+
+			// return ast.ObjectTerm(k), nil
```
Comment (marvinkite) [inline functions/identity.go:90]: If not used let's remove the dead code.

File: main.go [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+		os.Exit(1)
+	}
+	// log "github.com/sirupsen/logrus"
+	// joonix "github.com/joonix/log"
+	// For GCP log.SetFormatter(joonix.NewFormatter())
```
Comment (marvinkite) [inline main.go:35]: Let's remove these comments.

File: plugins/identity.go [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+	manager *plugins.Manager
+	config  *Config
+	client  *identity.Client
+	mtx     sync.Mutex
+}
+
+func (p *IndyKitePlugin) Start(ctx context.Context) (err error) {
+	var options []api.ClientOption
+	if p.config.UseEnvVariables {
+		//	api.WithCredentialsLoader(api.StaticCredentialConfig(p.config.credConfig)),
```
Comment (marvinkite) [inline plugins/identity.go:103]: Let's remove this comment

**Review by marvinkite — CHANGES_REQUESTED** [review]
Looks good, there are some clean up comments.

File: README.md (hunk 1 above)
Comment (arxeiss) [reply README.md:9]: Yes, fixed

File: utilities/conversion.go [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+		return ast.UIntNumberTerm(pv.UnsignedIntegerValue), nil
+	case *objects.Value_DoubleValue:
+		return ast.FloatNumberTerm(pv.DoubleValue), nil
+	case *objects.Value_ValueTime:
+		if pv.ValueTime == nil {
+			return ast.NullTerm(), nil
+		}
+		return ast.StringTerm(pv.ValueTime.AsTime().Format(time.RFC3339)), nil
+	// case *objects.Value_DurationValue:
+	// 	a := pv.DurationValue.
```
Comment (maaland) [inline utilities/conversion.go:68]: remove comment

**Review by maaland — COMMENTED** [review]
I agree with the others' comments.

File: main.go (hunk 4 above)
Comment (arxeiss) [reply main.go:30]: Yep, required

File: config.yaml (hunk 3 above)
Comment (arxeiss) [reply config.yaml:58]: No, will clean up

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (arxeiss) [reply README.md:9]: Sure, fixed to `indykite/opa`

File: functions/client.go (hunk 6 above)
Comment (arxeiss) [reply functions/client.go:40]: Fixed and added linter to force this

File: functions/client.go (hunk 5 above)
Comment (arxeiss) [reply functions/client.go]: Done and force with linter too

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+<!-- _Please make sure to review and check all of these items:_ -->
+
+# Checklist
+
+<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->
+
+- [ ] `make test` passes
+- [ ] tests and/or benchmarks are included
+- [ ] documentation is changed or added
+- [ ] commit message follows [commit guidelines](https://github.com/indykite/jarvis-sdk-go/blob/master/doc/guides/commit-message.md)
```
[the remark this answers is not in the document]
Comment (arxeiss) [reply .github/PULL_REQUEST_TEMPLATE.md]: Fixed and create PR with updated file in organization level repo

**Review by maaland — APPROVED** [review]

**Review by marvinkite — APPROVED** [review]

**Pull request merged**